### DISCUSSION TITLE
suddenly getting `Could not identify Next.js version` error

### DISCUSSION THREAD

**Discussion by rabelais88** [vercel/community#1391] [Help] [answered]
# Issue
the very same project deployed successfully [42days ago](https://vercel.com/rabelais88/sungryeol/HRjoModM91TSujHgvvMba8g2GNYo) but now it is getting `Could not identify Next.js version` error.

there have been some minor changes between builds; none of them included deploy or package related updates but somehow the error suddenly appears out of nowhere.

## latest build fail logs
```


Cloning github.com/rabelais88/sungryeol (Branch: main, Commit: 50534cd)
--
23:48:34.901 | Skipping build cache since Vercel CLI used --force option
23:48:36.199 | Cloning completed: 1.305s
23:48:36.199 | Running "bash check-build.sh"
23:48:36.281 | VERCEL_GIT_COMMIT_MESSAGE: reinstall for solving cache issues + added vercel build command to readme
23:48:36.282 | ✅ - Build can proceed
23:48:36.389 | Running "vercel build"
23:48:36.789 | Vercel CLI 28.13.2
23:48:36.988 | Running "install" command: `pnpm install -r`...
23:48:37.427 | No projects found in "/vercel/path0/app-portfolio-2022"
23:48:37.433 | Warning: Could not identify Next.js version, ensure it is defined as a project dependency.
23:48:37.481 | Error: No Next.js version could be detected in your project. Make sure `"next"` is installed in "dependencies" or "devDependencies"


```

## project detail

- monorepo + single app deploy(install all packages -> deploy a single private app from the repo) + turborepo
- the app name: `@sungryeol/app-portfolio-2022`
- the app folder: `./app-portfolio-2022`
- pnpm

- build command
```sh
cd .. && npx turbo run deploy --filter=@sungryeol/app-portfolio-2022
```

- install command
```sh
pnpm install -r
```

- node version 18.x
- root folder: `app-portfolio-2022`
- Included source files outside of the Root Directory in the Build Step

- turborepo.json
```json
{
  "$schema": "https://turborepo.org/schema.json",
  "pipeline": {
    "prebuild": {},
    "build": {
      "dependsOn": ["prebuild", "^build"],
      "outputs": [".next/**", "dist/**"]
    },
    "test": {
      "dependsOn": ["build"],
      "inputs": ["src/**/*.tsx", "src/**/*.ts", "test/**/*.ts", "test/**/*.tsx"]
    },
    "deploy": {
      "dependsOn": ["prebuild", "build", "sitemap"],
      "outputs": []
    },
    "sitemap": {
      "dependsOn": ["build"],
      "outputs": ["public/robots.txt", "public/sitemap*.xml"]
    },
    "dev": {
      "cache": false
    }
  },
  "globalDependencies": [
    "typescript-config/tsconfig.json",
    "prettier-config/index.js"
  ]
}
```

## what I've tried
- updated next.js to the latest version: 12.x -> 13.x
- redeployed without cache
- changed install command: `pnpm install` -> `pnpm install -r`

**Comment by amyegan** [comment]
Thanks for reaching out @rabelais88. The first thing I want to check is whether the problem is related to the CLI version. Try adding an environment variable in your project dashboard with the name `VERCEL_CLI_VERSION` and value `vercel@28.9.0` and then redeploy. 

If that works, it will give us some clues about the source of the problem. Just remember that this is only a temporary workaround for debugging and the environment variable need to be removed after we find a real solution! Your project will be stuck on an outdated CLI version as long as you leave this env var in place, which could cause more problems in the future.

**Reply by rabelais88** [reply]
@amyegan the build still fails. [to the latest build log](https://vercel.com/rabelais88/sungryeol/64Musbqp5msdo3823Gk8h47vMzJt)

- build log
```


Cloning github.com/rabelais88/sungryeol (Branch: main, Commit: d3e8236)
--
10:39:36.595 | Skipping build cache since Vercel CLI used --force option
10:39:38.116 | Cloning completed: 1.526s
10:39:38.116 | Running "bash check-build.sh"
10:39:38.220 | VERCEL_GIT_COMMIT_MESSAGE: update, vercel version 28.4.8 -> 28.14.0, turbo version 1.5.5 -> 1.7.0
10:39:38.220 | ✅ - Build can proceed
10:39:38.376 | Installing Vercel CLI: vercel@28.14.0
10:39:46.223 | Running "vercel build"
10:39:46.663 | Vercel CLI 28.14.0
10:39:46.890 | Running "install" command: `pnpm install -r`...
10:39:47.384 | No projects found in "/vercel/path0/app-portfolio-2022"
10:39:47.390 | Warning: Could not identify Next.js version, ensure it is defined as a project dependency.
10:39:47.451 | Error: No Next.js version could be detected in your project. Make sure `"next"` is installed in "dependencies" or "devDependencies"


```
- updated vercel version 28.4.8 -> 28.14.0, turbo version 1.5.5 -> 1.7.0
- the build fails with or without `VERCEL_CLI_VERSION` env var.

**Reply by rabelais88** [reply]
I've also noticed that changing package version does not trigger an install. It always says `Already up-to-date`, even when there's an explicit change in package versions.

```
Running "install" command: `pnpm install`...
--
10:41:45.783 | Already up-to-date
```

**Reply by amyegan** [reply]
Thanks trying that. I see the build log you shared used Vercel CLI 28.14.0. Did you have a build with Vercel CLI 28.9.0? That looks like the version used for your most recent successful deployment.

Have you changed your Build & Development Settings, Root Directory, or any other settings since the last successful deployment?

**Reply by rabelais88** [reply]
@amyegan that's actually a redeploy from the first failure. Vercel version did not affect build success. I have already tried the mentioned stuff:
- using original Vercel version(`vercel@28.9.0`): initial build failure happened without any Vercel infrastructure changes. The log that I included was the latest error which includes different version for fix, did not work.
- isolated build(without root directory)
- redeploy without cache
- added `vercel.json` setting
- changing turbo deploy command(`npx turbo run deploy` -> default vercel deploy command)
- changing install command

and it is weird because Vercel does not seem to acknowledge the project folder settings at all and pnpm installation always say every library is "up to date", even when there is a noticeable change in npm libs. I think there might be something wrong with new Vercel install script, cache system, or project folder recognition script.

in case it is necessary, here's [the git repository link](https://github.com/rabelais88/sungryeol)

**Reply by amyegan** [reply]
Thanks for clarifying. These details are good to know, and I've reached out to the engineering team with this info to get their help

**Reply by rabelais88** [reply]
thanks for your help, @amyegan !

**Comment by Ethan-Arrowood** [comment, marked as answer]
Hi @rabelais88 please update your `.vercelignore` to the correct path

```diff
/*
- !app-portfolio-2021
+ !app-portfolio-2022
```

**Reply by rabelais88** [reply]
@Ethan-Arrowood Thank you. Now it works. How foolish of me to ignore this.
